engine.py

```diff
@@ -28,9 +28,42 @@
 
 BASEPATH = os.path.dirname(os.path.abspath(__file__))
 
-
-class Engine():
+if os.name == 'nt':
+    WINDOWS = True
+else:
+    WINDOWS = False
+
+if WINDOWS:
+    import pydivert
+    from pydivert.consts import Direction
+else:
+    import netfilterqueue
+
+from abc import ABC, abstractmethod
+
+def Engine(server_port, string_strategy, environment_id=None, output_directory="trials", log_level="info"):
```

Comment:
can this just take args and **kwargs? makes it easier to extend one or both engines going forward (and for merging in the additional engine flags coming down the pipe)